micro-rdk/src/common/registry.rs

```diff
@@ -133,6 +133,7 @@ type GenericComponentConstructor =
 
 type DependenciesFromConfig = dyn Fn(ConfigType) -> Vec<ResourceKey>;
 
+#[derive(Clone)]
```

Comment:
why does this now need to be cloneable?

Reply:
I agree it's not ideal. We should keep track of this with a ticket if we want to leave this as is when merging

Reply:
https://viam.atlassian.net/browse/RSDK-8122